File: 108.py
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        n = len(nums)
        if n == 0:
            return None
        return TreeNode(
            nums[n // 2],
            self.sortedArrayToBST(nums[:n // 2]),
            self.sortedArrayToBST(nums[n // 2 + 1:])
        )

File: sol/108.cpp
struct Solution {
    TreeNode *gen(int i, int j, vector<int>& nums){
        if (i >= j){
            return nullptr;
        }
        int m = (i + j) / 2;
        return new TreeNode(
                nums[m],
                this->gen(i, m, nums),
                this->gen(m + 1, j, nums)
        );
    }
    inline TreeNode* sortedArrayToBST(vector<int>& nums) {
        return this->gen(0, nums.size(), nums);
    }
};
